src/core/thread/mle.cpp

```diff
@@ -1713,43 +1713,67 @@ void Mle::HandleAttachTimer(Timer &aTimer)
     aTimer.Get<Mle>().HandleAttachTimer();
 }
 
-void Mle::HandleAttachTimer(void)
+bool Mle::DidFindAcceptableParentCandidate(void) const
 {
-    uint32_t delay          = 0;
-    bool     shouldAnnounce = true;
+    bool didFind = false;
 
-    if (mAttachState == kAttachStateParentRequestRouter || mAttachState == kAttachStateParentRequestReed ||
-        (mAttachState == kAttachStateAnnounce && !HasMoreChannelsToAnnouce()))
+    switch (mAttachState)
     {
-        uint8_t linkQuality;
+    case kAttachStateParentRequestRouter:
+    case kAttachStateParentRequestReed:
+        break;
 
-        linkQuality = mParentCandidate.GetLinkInfo().GetLinkQuality();
+    case kAttachStateAnnounce:
+        VerifyOrExit(!HasMoreChannelsToAnnouce());
+        break;
```

Comment:
```suggestion
    case kAttachStateAnnounce:
        VerifyOrExit(!HasMoreChannelsToAnnouce());
```
nit: can remove `break` after moving `case kAttachStateAnnounce` to the beginning of the switch.

Reply:
Changed in new push. We need `OT_FALL_THROUGH` instead of `break`. But I agree it reads a bit better (to have all states that we allow to map to single/same `break`)